junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/DiscoverySelectorResolver.java

```diff
@@ -64,7 +64,7 @@ public void resolveSelectors(EngineDiscoveryRequest request, TestDescriptor engi
 		request.getSelectorsByType(UniqueIdSelector.class).forEach(selector -> {
 			javaElementsResolver.resolveUniqueId(selector.getUniqueId());
 		});
-		pruneTree(engineDescriptor);
+		engineDescriptor.pruneTree();
```

Comment:
I don't think anyone is depending on this assumption but let's move this investigation into a separate issue. Can you create one, please?